src/app/store/effects/pokemon.effects.ts

```diff
@@ -0,0 +1,155 @@
+import { selectFromStore } from '../selectors/pokemon.selectors';
+
+
+@Injectable()
+export class PokemonEffects {
+
+  loadPokemons$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.getPokemonFromApi),
+      tap(
+        () => {
+          this.listService.getAllPokemons();
+        })
+    ), { dispatch: false });
+
+  showPokemon$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.loadPokemonList),
+      tap(
+        () => {
+          this.store.pipe(
+            select(selectFromStore),
+            first()
+          ).subscribe(
+            (state) => {
+              this.store.dispatch(setPokemonsToShow({ pokemonsToShow: state.pokemonList }));
+            }
+          )
+        })
+    ), { dispatch: false });
+
+  loadSelectedPokemon$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.selectPokemonCard),
+      tap((action) => {
+        this.service.loadSelectedPokemon(action.name);
+      })
+    ), { dispatch: false });
+
+  openModal$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.showPokemonDetails),
+      tap(
+        () => {
+          this.dialog.open(DialogPokemonInfoComponent,
+            { panelClass: 'custom-dialog-container' });
+        })
+    ), { dispatch: false });
+
+  activateComparation$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.selectToCompare),
+      tap(() => {
+        this.service.setCompareTo();
+      })
+    ), { dispatch: false });
+
+  openComparationModal$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.comparePokemonTo),
+      tap(
+        () => {
+          this.dialog.open(DialogPokemonComparationComponent,
+            { panelClass: 'custom-dialog-container' });
+          this.service.setCompareTo();
+        })
+    ), { dispatch: false });
+
+  lookForPokemons$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.lookFor),
+      tap(
+        (action) => {
+          this.store.pipe(
+            select(selectFromStore),
+            first()
+          ).subscribe(
+            state => {
+              if (action.word.length > 0) {
+                this.store.dispatch(setPokemonsToShow({
+                  pokemonsToShow: state.pokemonList.filter(
+                    pokemon => pokemon.name.startsWith(action.word)
+                  )
+                }));
+              } else {
+                this.store.dispatch(setPokemonsToShow({
+                  pokemonsToShow: state.pokemonList
+                }));
+              }
+            }
+          );
+        })
+    ), { dispatch: false });
+
+  addToFavorites$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.favorite),
+      tap(
+        (action) => {
+          this.store.pipe(
+            select(selectFromStore),
+            first()
+          ).subscribe(
```

Comment:
in theory the effects are being subscribe when they are created, therefore this is a subscribe inside a subscribe and must be removed, use switchmap/mergemap isntead